2015-10-28  Connor Robert Kenny  <connork@lnxsrv04.seas.ucla.edu>

	* timespec.h: Change how timespec.h compares the times to prevent some really old dates from overflowing.
	This is done by comparing with < and > instead of subtracting

--- coreutils-with-bug/lib/timespecOLD.h	2015-10-28 11:15:44.868484000 -0700
+++ coreutils-with-bug/lib/timespec.h	2015-10-28 11:06:48.632310000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int diff = 0;
+  if (a.tv_sec < b.tv_sec)
+    diff = -1;
+  else if (a.tv_sec > b.tv_sec)
+    diff = 1;
+  return diff;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
